pkg/storage/backend-storage/backend-storage.go

```diff
@@ -74,21 +153,95 @@ func IsBackendStorageNeededForVM(vm *corev1.VirtualMachine) bool {
 	return HasPersistentTPMDevice(&vm.Spec.Template.Spec)
 }
 
+func MigrationHandoff(client kubecli.KubevirtClient, pvcStore cache.Store, migration *corev1.VirtualMachineInstanceMigration) error {
+	if migration == nil || migration.Status.MigrationState == nil ||
+		migration.Status.MigrationState.SourcePersistentStatePVCName == "" ||
+		migration.Status.MigrationState.TargetPersistentStatePVCName == "" {
+		return fmt.Errorf("missing source and/or target PVC name(s)")
+	}
+
+	sourcePVC := migration.Status.MigrationState.SourcePersistentStatePVCName
+	targetPVC := migration.Status.MigrationState.TargetPersistentStatePVCName
+
+	if sourcePVC == targetPVC {
+		// RWX backend-storage, nothing to do
+		return nil
+	}
+
+	// Let's label the target first, then remove the source.
+	// The target might already be labelled if this function was already called for this migration
+	target := pvcForMigrationTargetFromStore(pvcStore, migration)
+	if target == nil {
+		return fmt.Errorf("target PVC not found for migration %s/%s", migration.Namespace, migration.Name)
+	}
+	labels := target.Labels
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+
+	existing, ok := labels[PVCPrefix]
+	if ok && existing != migration.Spec.VMIName {
+		return fmt.Errorf("target PVC for %s is already labelled for another VMI: %s", migration.Spec.VMIName, existing)
+	}
+
+	labelPatch := patch.New()
+	labelPatch.AddOption(patch.WithReplace("/metadata/labels/"+PVCPrefix, migration.Spec.VMIName))
+	labelPatch.AddOption(patch.WithRemove("/metadata/labels/" + patch.EscapeJSONPointer(corev1.MigrationNameLabel)))
+
+	labelPatchPayload, err := labelPatch.GeneratePayload()
+	if err != nil {
+		return fmt.Errorf("failed to generate PVC patch: %v", err)
+	}
+	_, err = client.CoreV1().PersistentVolumeClaims(migration.Namespace).Patch(context.Background(), targetPVC, types.JSONPatchType, labelPatchPayload, metav1.PatchOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to patch PVC: %v", err)
+	}
```

Comment:
I think we have to cover the case where the patch of the targetPVC succeeds but the subsequent deletion fails.
As is, in the second iteration when we try to patch the targetPVC again it will fail because the `corev1.MigrationNameLabel` was already gone.
Or am I missing something?